exonum-java-binding/core/src/test/java/com/exonum/binding/runtime/ServiceRuntimeTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.runtime;
+
+import static com.exonum.binding.runtime.FrameworkModule.SERVICE_WEB_SERVER_PORT;
+import static com.exonum.binding.runtime.ServiceRuntimeTest.PORT;
+import static com.google.inject.name.Names.named;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.exonum.binding.service.adapters.UserServiceAdapter;
+import com.exonum.binding.service.adapters.ViewFactory;
+import com.exonum.binding.transport.Server;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Singleton;
+import com.google.inject.Stage;
+import java.nio.file.Paths;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * An integration test of ServiceRuntime + Guice interop.
+ */
+@ExtendWith(MockitoExtension.class)
+class ServiceRuntimeTest {
+
+  static final int PORT = 25000;
```

Comment:
it can be a situation when the port is taken (hopefully really rare). 
How about to make `com.exonum.binding.transport.Server#start` returns actual port, or extend the interface with `getActualPort` method? Currently, if we pass `0` as a port to the `Server` it will run on a random port and there is no way to get it.

Reply:
I see, that is a sensible thing to do for integration tests that use `VertxServer`. In production, however, the server must either start on the configured port or fail. I think we can discuss it in a separate issue: https://jira.bf.local/browse/ECR-3016 

These tests use a mock of `Server`, so in this case it is totally reliable.